projects/hathor-wallet-mobile/0002-nano-contract-integration.md

```diff
@@ -0,0 +1,476 @@
+  
+## Registering Nano Contract
+
+```mermaid
+graph LR
+    A[Main] -->B[Nano Contract Transactions]
+    A -->|This Flow| C[Nano Contract Registration]
+```
+
+### Screen: Nano Contract Registration
+
+Users arrive at this screen from the *Main* screen to register a new Nano Contract. The screen features a form with an input field for *Nano Contract ID* and a *Register* button.
+
+Upon entering a valid *Nano Contract ID* and tapping *Register*, users receive immediate feedback based on the action's outcome: loading, success, or failure.
+
+Upon successful registration, users are redirected to the *Main* screen  the with *Nano Contract List* active.
+
+##### Design Bill
+- **Placement**:
+	- Place the form input on the top of the screen.
+	- Place the action button on the bottom of the screen.
+
+- **Functionality**:
+	- Upon tapping *Register*, initiate the registration process with appropriate loading feedback.
+
+- **Visibility Conditions**:
+	- The *Register* button remains inactive until a valid Nano Contract ID is entered into the input field.
+	- In case of processing success, display a success feedback and navigate back to the *Main* screen with the *Nano Contract List* active.
+
+- **Error Management**:
+	- In case of validation error, display an error message near the input field *Nano Contract ID*.
+	- In case of processing failure, display an error feedback.
+
+## Wallet Connect Interaction
+
+The Wallet Connect has a different *modus operandi*, it requires the user to interact with the dapp by modals.
+
+We may want to make the Wallet Connect requests from dapps assynchronous with push notification, for this read the alternative [Push notification for wallet connect requests](#push-notification-wallet-connect-requests).
+
+```mermaid
+graph LR
+    A[Any Screen] --> B[Nano Contract Transaction Request, Modal]
+```
+
+### Modal: Nano Contract Transaction Request
+[modal-nano-contract-transaction-request](#modal-nano-contract-transaction-request)
+
+This modal presents detailed information for a *Nano Contract Transaction Request* from dapp, allowing the user to approve or reject the transaction. It activates when a dapp makes a `htr_createNcTx` request to the wallet's RPC.
+
+The modal presents detailed information containing *Blueprint ID*, *Blueprint Method*, *Caller*, *Nano Contract ID* (if applicable), *Action List*, and *Argument List*. An *Action List* can be either types *deposit* or *withdrawal*. Both the lists can be empty.
+
+##### Design Bill
+- **Placement**:
+	- The modal must render centrally on top of the current screen, ensuring clear visibility and focus.
+	
+- **Functionality**:
+	- Include detailed transaction information for user review:
+		- *Blueprint ID*, *Blueprint Method*, *Caller*, *Nano Contract ID* (if applicable), *Action List*, and *Argument List*.
+	- Include detailed information for each *Action List* item:
+		- *Type*, *Token* and *Data*.
```

Comment:
What is data? Is it different for different nano contracts?

Reply:
Data encompass "address" and "amount" for an action type "withdrawal". If it is a "deposit" type, it will have only the "amount". In the end it will be translated to UTXOs.

Reply:
Are there other action types? Can you add more context in the design?

Reply:
No, there aren't more types than that. Sure, I will add more context in the design.